### PR TITLE
Updating to hydra-collections 8.0.0, fixes #1614

### LINKED ISSUE #1614: Get Sufia running with CC 0.9 and HC 8.0
New updates to Curation Concerns and Hydra Collections have broken the build resulting in these errors:
`wrong number of arguments (0 for 1)` when uploading files or creating works, and `You attempted to set the property `title' to a scalar value. However, this property is declared as being multivalued.` on collections.

### PR DESCRIPTION
**Pull request by awead**
Fixes #1614 

Updates Sufia to hydra-collections 8.0 and includes some test refactoring.

Description can have multiple paragraphs and you can use code examples inside:

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: app/models/concerns/sufia/collection_behavior.rb [hunk 1]
```diff
@@ -3,8 +3,9 @@ module CollectionBehavior
     extend ActiveSupport::Concern
 
     included do
+      include CurationConcerns::CollectionBehavior
```
Comment (jcoyne) [inline app/models/concerns/sufia/collection_behavior.rb]: Is this necessary? It's already generated into app/models/collection.rb, right?

File: app/presenters/sufia/collection_presenter.rb [hunk 2]
```diff
@@ -27,10 +25,12 @@ def [](key)
       end
     end
 
+    # TODO: Remove? this is in CC, see #1620
```
Comment (jcoyne) [inline app/presenters/sufia/collection_presenter.rb]: Yes

File: app/presenters/sufia/collection_presenter.rb [hunk 3]
```diff
@@ -27,10 +25,12 @@ def [](key)
...
     def size
       number_to_human_size(Sufia::CollectionSizeService.run(solr_document))
     end
 
+    # TODO: Remove? this is in CC, see #1620
```
Comment (jcoyne) [inline app/presenters/sufia/collection_presenter.rb]: Yes, remove

File: app/models/concerns/sufia/collection_behavior.rb (hunk 1 above)
Comment (awead) [inline app/models/concerns/sufia/collection_behavior.rb]: Yes, it is generated. I'll remove it.

File: app/presenters/sufia/collection_presenter.rb [hunk 4]
```diff
@@ -27,10 +25,12 @@ def [](key)
       end
     end
 
+    # TODO: Remove? this is in CC, see #1620
```
Comment (jcoyne) [inline app/presenters/sufia/collection_presenter.rb]: Yes, remove this.

**Comment by awead** [comment]
@jcoyne I removed those methods, but the build failed. I had noted that we should remove these methods in a follow up issue #1620. Can we merge this PR now? Others are waiting on a green build in order to merge their PRs @hectorcorrea.

**Comment by awead** [comment]
Also, to note, the follow up PR would also include removing duplicate views and other extraneous code.

**Comment by cam156** [comment]
:+1: To getting the broken build working

**Comment by mjgiarlo** [comment]
:clap: for :green_heart: build!

**Comment by cbeer** [comment]
Do we need to bump the gemspec requirements?

**Comment by mjgiarlo** [comment]
@cbeer I suspect @awead didn't change that because CurationConcerns is already pinning this dependency: https://github.com/projecthydra-labs/curation_concerns/blob/master/curation_concerns-models/curation_concerns-models.gemspec#L23

Do you think Sufia should declare it as well?

**Comment by cbeer** [comment]
We could bump the `CurationConcerns` dependency. Right now, it's `spec.add_dependency 'curation_concerns', '~> 0.7'`.

**Comment by mjgiarlo** [comment]
:+1: to that, @cbeer

**Comment by cbeer** [comment]
:ok_hand:

**Comment by awead** [comment]
yeah, sorry @cbeer, missed that. if you want to submit a PR, i'll merge it.

**Pull request merged**